application/src/objects/car-factory.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+var CarSprite = require('./car-sprite');
+
+var CarFactory = function() {};
+
+CarFactory.prototype.loadAssets = function(state)
+{
+    state.load.image('car', this.getSpritePath());
+};
+
+CarFactory.prototype.getSpritePath = function()
+{
+    return 'assets/img/bluebox.png';
+};
+
+CarFactory.prototype.spritePrototype = CarSprite;
+
+CarFactory.prototype.getSprite = function(state, x, y, key, group)
```

Comment:
Could we pass the state object into the factory constructor instead of having to pass it into every call to `getSprite`?